File: java/122A.java
// 122 A. Lucky Division

import java.util.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        long n = sc.nextLong();
        int[] lucky = {
            4, 7, 44, 47, 74, 77,
            444, 447, 474, 477, 744, 747, 774, 777
        };
        for (int x : lucky) {
            if (n % x == 0) {
                System.out.println("YES");
                return;
            }
        }
        System.out.println("NO");
    }
}

File: cpp/122A.cpp
// 122 A. Lucky Division

#include <bits/stdc++.h>
using namespace std;

int main() {
    long long n; 
    cin >> n;
    vector<int> lucky = {
        4, 7, 44, 47, 74, 77,
        444, 447, 474, 477, 744, 747, 774, 777
    };
    for (int x : lucky) {
        if (n % x == 0) {
            cout << "YES";
            return 0;
        }
    }
    cout << "NO";
    return 0;
}
